product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Student.scala

```diff
@@ -21,6 +22,54 @@ case class Student(username: String,
     copy(username.trim, role, address.trim, firstName.trim, lastName.trim,
       picture.trim, email.trim, birthDate.trim, immatriculationStatus.trim, matriculationId.trim)
   }
+
+  /** @inheritdoc */
+  override def validate: Seq[SimpleError] = {
+    val fos = List("Computer Science","Philosophy","Media Sciences", "Economics", "Mathematics", "Physics", "Chemistry",
+      "Education", "Sports Science", "Japanology", "Spanish Culture", "Pedagogy", "Business Informatics", "Linguistics")
+
+    var errors = List[SimpleError]()
+    errors ++= super.validate
+    if(!(matriculationId forall Character.isDigit) || !(matriculationId.toInt > 0)) {
+      errors :+= SimpleError("matriculationId", "Student ID invalid.")
+    }
+    if(!(semesterCount > 0)) {
+      errors :+= SimpleError("semesterCount", "Semester count must be a positive integer.")
+    }
+    if(!fieldsOfStudy.forall(fos.contains)) {
+      errors :+= SimpleError("fieldsOfStudy", "Fields of Study must be one of [..].")
+    }
+    errors
+  }
+
+
+  /** 
+    * Compares the object against the user parameter to find out if fields, which should only be changed by users with elevated privileges, are different.
+    * Returns a list of SimpleErrors[[de.upb.cs.uc4.shared.client.SimpleError]]
+    * 
+    * @param user 
+    * @return Filled Sequence of [[de.upb.cs.uc4.shared.client.SimpleError]]
+    */
+  def checkEditableFields (user: Student): Seq[SimpleError] = {
+
+    var errors = List[SimpleError]()
+   
+    errors ++= super.checkEditableFields(user)
```

Comment:
You could combine these three lines into one for better style and a little bit more efficiency